### PR TITLE
Improve relcoation logs

### PR DESCRIPTION
**Pull request by joshuef**

### CODE REVIEW

File: sn_node/src/node/messaging/node_msgs.rs [hunk 1]
```diff
@@ -178,7 +178,8 @@ impl MyNode {
                 let mut node = node.write().await;
                 trace!("[NODE WRITE]: RelocationRequest write gottt...");
                 trace!("Handling RelocationRequest msg from {sender}: {msg_id:?}");
-                Ok(node.handle_relocation_request(relocation_node, relocation_trigger)?)
+                Ok(node
+                    .handle_relocation_request_from_member(relocation_node, relocation_trigger)?)
```
Comment (oetyng) [inline sn_node/src/node/messaging/node_msgs.rs:182]: > handle_relocation_request_from_member
can there be such requests from non-members?

File: sn_node/src/node/mod.rs [hunk 2]
```diff
@@ -337,6 +337,18 @@ mod core {
             ))
         }
 
+        /// Generates a AE probe for our section elders
+        pub(crate) fn generate_section_probe_msg(context: &NodeContext) -> Cmd {
+            // Get prefixes of other sections.
```
Comment (oetyng) [inline sn_node/src/node/mod.rs:342]: Is this a copy paste comment? Doesn't seem to match the code.

File: sn_node/src/node/flow_ctrl/periodic_checks.rs [hunk 3]
```diff
@@ -272,38 +287,30 @@ impl FlowCtrl {
...
-    //             issue: sn_fault_detection::IssueType::AeProbeMsg,
-    //         });
-    //     }
+        // Send a probe message to an elder
+        debug!("Going to health check elders");
+
+        let elders = context.network_knowledge.elders();
+        for elder in elders {
+            // we track a knowledge issue
+            // whhich is countered when an AE-Update is
```
Comment (oetyng) [inline sn_node/src/node/flow_ctrl/periodic_checks.rs:302]: whhich => which

File: sn_node/src/node/flow_ctrl/periodic_checks.rs [hunk 4]
```diff
@@ -131,9 +150,15 @@ impl FlowCtrl {
                         NodeMsg::TryJoin(Some(proof.clone())),
                     ));
                 } else {
```
Comment (oetyng) [inline sn_node/src/node/flow_ctrl/periodic_checks.rs:152]: I always try to keep number of negations as few as possible, so the if clause here would be reversed to

`if context.network_knowledge.is_section_member(&context.name) {`

with line 153-163 in it, and current ln 145-151 within the `else` clause.

File: sn_node/src/node/messaging/node_msgs.rs (hunk 1 above)
Comment (joshuef) [reply sn_node/src/node/messaging/node_msgs.rs:182]: nodes joining from other sections was what I thought this was doing. Rename is for clarity on that

Comment (oetyng) [reply sn_node/src/node/messaging/node_msgs.rs:182]: I think they are handled via the join flow?

Looking that up now.

Comment (joshuef) [reply sn_node/src/node/messaging/node_msgs.rs:182]: > was supposed to be used as a verifiable proof of the trigger actually giving the relocation of that node

Was it? I thought it was just that the trigger was verifiable from the churnId now? cc @b00f , what's the case there?

Comment (oetyng) [reply sn_node/src/node/messaging/node_msgs.rs:182]: It was another issue, that's why I removed that part of the comment.

So, the refactor that introduced the node driving the process did not include the change to have the trigger verifiable.

Comment (oetyng) [reply sn_node/src/node/messaging/node_msgs.rs:182]: I'm raising a PR to clarify this flow with some name updates.

Comment (b00f) [reply sn_node/src/node/messaging/node_msgs.rs:182]: BeginRelocating and RelocationRequest sounds a bit wired for me also. I think we need to give a better name (if we can't merge them). By the way, relocation trigger has an instance of the decision that is the proof. AFAIK, to verify relocation we need the trigger. So here both items have triggers.

File: sn_node/src/node/flow_ctrl/periodic_checks.rs [hunk 5]
```diff
@@ -269,38 +283,30 @@ impl FlowCtrl {
...
-    //             issue: sn_fault_detection::IssueType::AeProbeMsg,
-    //         });
-    //     }
+        // Send a probe message to an elder
+        debug!("Going to health check elders");
+
+        let elders = context.network_knowledge.elders();
+        for elder in elders {
+            // we track a knowledge issue
+            // which is countered when an AE-Update is
```
Comment (b-zee) [inline sn_node/src/node/flow_ctrl/periodic_checks.rs:298]: This comment doesn't seem to finish the

Comment (b-zee) [reply sn_node/src/node/flow_ctrl/periodic_checks.rs:298]: Sorry, this seems like it was already in the code. I thought you added it 😀

File: sn_node/src/node/flow_ctrl/periodic_checks.rs [hunk 6]
```diff
@@ -97,7 +97,16 @@ impl FlowCtrl {
     async fn enqueue_cmds_for_node_periodic_checks(&mut self, context: &NodeContext) -> Vec<Cmd> {
         let mut cmds = vec![];
 
-        // check if we have been asked to prepare for relocation
+        // here we specifically ask for AE prob msgs and manually
+        // track faults for missing AE, and theerefore hopefully keep members updated...
```
Comment (b-zee) [inline sn_node/src/node/flow_ctrl/periodic_checks.rs:101]:
```suggestion
        // track faults for missing AE, and therefore hopefully keep members updated...
```

**Review by b-zee — APPROVED** [review]

**Review by b-zee — APPROVED** [review]

File: sn_node/src/node/flow_ctrl/mod.rs [hunk 7]
```diff
@@ -75,6 +75,7 @@ pub(crate) struct FlowCtrl {
     cmd_sender_channel: Sender<(Cmd, Vec<usize>)>,
     fault_channels: FaultChannels,
     timestamps: PeriodicChecksTimestamps,
+    started_relocating: Option<XorName>,
```
Comment (RolandSherwin) [inline sn_node/src/node/flow_ctrl/mod.rs:78]: Sorry, I can't really figure out what the above field is doing. Is this a placeholder for a future change?

Comment (joshuef) [reply sn_node/src/node/flow_ctrl/mod.rs:78]: in periodic checks, we log when we start a relocation process (only the first time)

File: sn_node/src/node/flow_ctrl/cmds.rs [hunk 8]
```diff
@@ -203,8 +203,6 @@ pub(crate) enum Cmd {
 
 impl Cmd {
```
Comment (RolandSherwin) [inline sn_node/src/node/flow_ctrl/cmds.rs:204]: Please ignore this. Testing github auto-merge

Comment (RolandSherwin) [inline sn_node/src/node/flow_ctrl/cmds.rs:204]: Please ignore this as well. Testing github auto-merge

**Pull request merged**